src/components/PassphraseView.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useEffect } from 'react'
+import PassphraseForm from 'components/PassphraseForm'
+import Passphrase2FA from 'components/2FA/Passphrase2FA'
+import classNames from 'classnames'
+import viewStyles from 'styles/view'
+
+const PasswordView = props => {
+  const {
+    onPassphraseSimpleSubmit,
+    onPassphrase2FAStep1,
+    onPassphrase2FAStep2,
+    passphrase,
+    isTwoFactorEnabled,
+    instance,
+    fetchInfos
+  } = props
+
+  useEffect(() => {
+    fetchInfos()
+  }, [])
+
+  const [twoFAModalOpen, setTwoFAModalOpen] = useState(false)
+  const [currentPassphrase, setCurrentPassphrase] = useState('')
+  const [new2FAPassphrase, setNew2FAPassphrase] = useState('')
+
+  const handlePassphrase2FAStep1 = (current, newVal) => {
+    setCurrentPassphrase(current)
+    setNew2FAPassphrase(newVal)
+    setTwoFAModalOpen(true)
+    onPassphrase2FAStep1(current)
+  }
+
+  const handlePassphrase2FASubmit = twoFactorCode => {
+    const { twoFactorToken } = passphrase
+    onPassphrase2FAStep2(
+      currentPassphrase,
+      new2FAPassphrase,
+      twoFactorCode,
+      twoFactorToken
+    ).then(() => setTwoFAModalOpen(false))
```

Comment:
It seems the UI is reacting after a network round trip but no visual indication is given to the user, I see no setPassphrase2FAStepSubmitting ? Is it expected ?